expression/builtin_other.go

```diff
@@ -79,7 +79,12 @@ func (c *inFunctionClass) getFunction(ctx sessionctx.Context, args []Expression)
 	bf.tp.Flen = 1
 	switch args[0].GetType().EvalType() {
 	case types.ETInt:
-		sig = &builtinInIntSig{baseBuiltinFunc: bf}
+		inInt := builtinInIntSig{baseBuiltinFunc: bf}
+		err := inInt.buildHashMapForConstArgs(ctx)
```

Comment:
I think hash set is not always better than the brute force enumeration. The operation of calculating hash value might bring more overhead. For example, SparkSQL uses a threshold of the number of args of `IN` to decide whether to use a hash set. Maybe you can make a benchmark to find this threshold.

Reply:
yes, I am going to find the threshold.